content/articles/how-to-regulate-sessions-with-redis/index.md

```diff
@@ -0,0 +1,272 @@
+#### Example
+```bash
+redis 127.0.0.1:6379> ZADD names 0 Paul 
+(integer) 1 
+redis 127.0.0.1:6379> ZADD names 0 Greenwood 
+(integer) 1 
+redis 127.0.0.1:6379> ZADD names 0 Angel 
+(integer) 1 
+redis 127.0.0.1:6379> ZADD names 0 Angel 
+(integer) 0 
+redis 127.0.0.1:6379> ZRANGEBYSCORE names 0 1000  
+
+1) "Paul" 
+2) "Greenwood" 
+3) "Angel"
+```
+
+There are plenty of other commands that you can learn with Redis and they are really easy to follow just with your documentation. Visit https://redis.io/topics/data-types-intro
+
+### Redis Caching (Node.js )
+
+Here we are going to make a request to a GitHub API to get the number of repos for users, and we are going to store that number in our Redis cache. So Redis is basically key-value peers, it allows us to make fewer requests so it will cache the data so we don't have to keep making requests and it will speed up our application. Here we are not going to build a real application am just going to show you how to implement it, but you are going to see the benefit of it and you are going to learn the foundation to use it in your node.js application.
+
+#### Open any code editor of your choice and run npm init-y in the terminal to create a quick package.json
+```bash
+npm init-y
+
+```
+We are going to install a couple of dependencies
+we are going to be making our request with the GitHub API using node-fet
+also, install nodemon
+```bash
+npm i express node-fetch redis
+
+```
+npm i -D nodemon
+
+```bash
+const express = require('express');
+const fetch = require('node-fetch');
+const redis = require('redis');
+
+// enviroment variables 
+const PORT = process.env.PORT || 5000;
+const REDIS_PORT = process.env.REDIS_PORT || 6379;
+
+const client = redis.createClient(REDIS_PORT);
+
+const app = express();
+
+// Set response
+function setResponse(username, repos) {
+  return `<h2>${username} has ${repos} Github repos</h2>`;
+}
+
+// Make request to Github for data
+async function getRepos(req, res, next) {
+  try {
+    console.log('Fetching Data...');
+
+    const { username } = req.params;
+
+    const response = await fetch(`https://api.github.com/users/${username}`);
+
+    const data = await response.json();
+
+    const repos = data.public_repos;
+
+    // Set data to Redis
+    client.setex(username, 3600, repos);
+
+    res.send(setResponse(username, repos));
+  } catch (err) {
+    console.error(err);
+    res.status(500);
+  }
+}
+
+// Cache middleware
+function cache(req, res, next) {
+  const { username } = req.params;
+
+  client.get(username, (err, data) => {
+    if (err) throw err;
+
+    if (data !== null) {
+      res.send(setResponse(username, data));
+    } else {
+      next();
+    }
+  });
+}
+
+app.get('/repos/:username', cache, getRepos);
+
+app.listen(5000, () => {
+  console.log(`App listening on port ${PORT}`);
+});
+```
+
```

Comment:
This looks like an abrupt end.

We can make sure to give some insights on the code above. Add images if necessary.